### ISSUE TITLE
oomd: [FAILED] Failed to start testsuite-55.service

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#22030]
Version: 250-1

We are currently trying to get oomd package for Debian. Part of that effort is getting TEST-55-OOMD run via autopkgtest/upstream-ci.

The first issue I stumbled upon was #22029. After addressing that, the test still fails.

```
Sent message type=signal sender=org.freedesktop.systemd1 destination=n/a path=/org/freedesktop/systemd1/unit/testsuite_2d55_2dtestbloat_2eservice inta
Sent message type=signal sender=org.freedesktop.systemd1 destination=n/a path=/org/freedesktop/systemd1/unit/testsuite_2d55_2dtestbloat_2eservice inta
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/testsuite_2d55_2dtestbloat_2eservice interface=org.freedesktoa
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/testsuite_2d55_2dtestbloat_2eservice interface=org.freedesktoa
Sent message type=signal sender=org.freedesktop.systemd1 destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager mea
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager member=JobRemoved cookia
Bus private-bus-connection: changing state RUNNING → CLOSING
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/testsuite_2d55_2dtestbloat_2eservice interface=org.freedesktoa
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/testsuite_2d55_2dtestbloat_2eservice interface=org.freedesktoa
testsuite-55-testbloat.service: Failed to send unit change signal for testsuite-55-testbloat.service: Connection reset by peer
systemd-journald.service: Got notification message from PID 179 (FDSTORE=1)
systemd-journald.service: Added fd 44 (n/a) to fd store.
Bus private-bus-connection: changing state CLOSING → CLOSED
Got disconnect on private connection.
systemd-journald.service: Received EPOLLHUP on stored fd 80 (stored), closing.
Received SIGCHLD from PID 231 (bash).
Child 231 (bash) died (code=exited, status=1/FAILURE)
testsuite-55.service: Child 231 belongs to testsuite-55.service.
testsuite-55.service: Main process exited, code=exited, status=1/FAILURE
testsuite-55.service: Failed with result 'exit-code'.
testsuite-55.service: Service will not restart (restart setting)
testsuite-55.service: Changed start -> failed
varlink-32: Sending message: {"parameters":{"cgroups":[{"mode":"auto","path":"/system.slice/testsuite-55.service","property":"ManagedOOMSwap"},{"mode}
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.DBus.Properties member=PropertiesChangeda
testsuite-55.service: Job 77 testsuite-55.service/start finished, result=failed
[FAILED] Failed to start testsuite-55.service.
```

Full log + journal is attached
[journal.txt](https://github.com/systemd/systemd/files/7824177/journal.txt)
[log.txt](https://github.com/systemd/systemd/files/7824179/log.txt)

**Comment by mrc0mmand** [comment]
Ah, I see. In the test we override `DefaultMemoryPressureDurationSec=` by appending it to `/etc/systemd.oomd.conf`.

https://github.com/systemd/systemd/blob/cd933f14bd70d8311799972ca71280a733eb1d6a/test/units/testsuite-55.sh#L25

However, in Debian you ship a dropin, which sets it to 20s:

https://salsa.debian.org/systemd-team/systemd/-/blob/debian/experimental/debian/extra/systemd-oomd-defaults/oomd.conf.d/10-oomd-defaults.conf

This dropin in this case takes precedence. I guess we should configure `DefaultMemoryPressureDurationSec=` using a dropin as well, but with a higher priority.

**Comment by mrc0mmand** [comment]
Not quite resolved yet, see #22029.
